core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -438,15 +438,17 @@ RexNode toRex(RelInput relInput, Object o) {
     } else if (o instanceof Map) {
       Map map = (Map) o;
       final Map<String, Object> opMap = (Map) map.get("op");
+      if (map.containsKey("class")) {
+        opMap.put("class", map.get("class"));
+      }
       final RelDataTypeFactory typeFactory = cluster.getTypeFactory();
```

Comment:
What if `opMap` is null?

Reply:
@chunweilei I think even `opMap` is null, it does not matter. And for RexCall, it could not be null.

Reply:
@chunweilei  thanks for point out. better to  move it down inside `if(opMap != null)`

Reply:
@wangzzu It matters since it might throw NPE. It should put inside` if(opMap != null)` block.